canonical/react-components
# feat: Migrate Percy from CircleCI to Github Actions

## Done

- Updates React Components to use Github Actions for Percy testing instead of CircleCI. This is a react-components version of https://github.com/canonical/vanilla-framework/pull/5089 and https://github.com/canonical/vanilla-framework/pull/5095.
- I also updated the yarn/npm `clean` command to delete the `docs` directory that is generated by the `build-docs` command.

## QA

### QA steps

- Review [test PR](https://github.com/jmuzina/react-components/pull/1) from my fork. Verify that it causes a [Percy prepare workflow](https://github.com/jmuzina/react-components/actions/runs/9520708588/job/26246548342?pr=1) to trigger a [Percy snapshot workflow](https://github.com/jmuzina/react-components/actions/runs/9520711532), which creates a [status check on the PR](https://github.com/jmuzina/react-components/pull/1/checks?check_run_id=26246767137) with a [link to a Percy build](https://percy.io/bb49709b/test-vanilla-gha-migration/builds/34780805/changed/1901788616?browser=chrome&browser_ids=56&group_snapshots_by=similar_diff&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=1280) showing changes from the PR.
- Note that this PR deletes the CircleCI config so it will fail the CircleCI status check. Also, this PR will not cause a Percy build, as the CircleCI config has been removed and the Percy snapshots workflow (introduced by this PR) is not yet in the main branch. Review the changed files list to verify no visual changes are introduced. Once this PR is merged, we can disable the CircleCI integration.

### Percy steps

- No visual changes expected. This does not need to create a new react-components version as no components were updated.

## Fixes

Fixes: [WD-12705](https://warthogs.atlassian.net/browse/WD-12705)

[WD-12705]: https://warthogs.atlassian.net/browse/WD-12705?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
